plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TableSnapshot.java

```diff
@@ -304,23 +305,26 @@ private Stream<DeltaLakeTransactionLogEntry> getV2CheckpointTransactionLogEntrie
                         builder.accept(entry);
                         return;
                     }
-                    Location sidecar = checkpointFile.location().sibling("_sidecars").appendPath(entry.getSidecar().path());
-                    CheckpointEntryIterator iterator = new CheckpointEntryIterator(
-                            fileSystem.newInputFile(sidecar),
-                            session,
-                            fileSize,
-                            checkpointSchemaManager,
-                            typeManager,
-                            entryTypes,
-                            metadataEntry,
-                            protocolEntry,
-                            stats,
-                            parquetReaderOptions,
-                            checkpointRowStatisticsWritingEnabled,
-                            domainCompactionThreshold,
-                            partitionConstraint,
-                            addStatsMinMaxColumnFilter);
-                    stream(iterator).onClose(iterator::close).forEach(builder);
+                    // Sidecar files contains only ADD and REMOVE entry types. https://github.com/delta-io/delta/blob/master/PROTOCOL.md#v2-spec
+                    if (entryTypes.contains(ADD) || entryTypes.contains(REMOVE)) {
```

Comment:
Can you do an early return here if these conditions are not met, to avoid the large indentation block?